components/scream/scripts/test_all_scream.py

```diff
@@ -50,6 +50,12 @@ def __init__(self, cxx_compiler, f90_compiler, c_compiler, submit=False, paralle
         #  Sanity checks and helper structs setup  #
         ############################################
 
+        # Unless the user claims to know what he/she is doing, we setup the env.
+        expect(self._machine is not None, "Machine is now required by test_all_scream")
```

Comment:
What if we defaulted the mach to `linux-generic`? It would save a bunch of keystrokes for the user who runs on a generic machine. As a debug check, we could print the mach (or maybe even more options) inside self._init, so we can always check the logs in case something goes wrong.

Also, _very minor_: maybe `test-all-scream` rather than `test_all_scream`.